runtime/binding-mqtt-kafka/src/main/java/io/aklivity/zilla/runtime/binding/mqtt/kafka/internal/stream/MqttKafkaSessionFactory.java

```diff
@@ -4874,6 +4881,37 @@ private void onKafkaReset(
             delegate.doMqttReset(traceId, EMPTY_OCTETS);
         }
 
+        private void onKafkaWindow(
+            WindowFW window)
+        {
+            final long traceId = window.traceId();
+            final long authorization = window.authorization();
+            final long budgetId = window.budgetId();
+            final long sequence = window.sequence();
+            final long acknowledge = window.acknowledge();
+            final int maximum = window.maximum();
+            final boolean wasOpen = MqttKafkaState.initialOpened(state);
+
+            assert acknowledge <= sequence;
+            assert acknowledge >= delegate.initialAck;
+            assert maximum >= delegate.initialMax;
+
+            initialAck = acknowledge;
+            initialMax = maximum;
+            state = MqttKafkaState.openInitial(state);
+
+            assert initialAck <= initialSeq;
+
+            if (!wasOpen)
+            {
+                delegate.onOffsetCommitOpened(traceId, authorization, budgetId);
+            }
+            else
+            {
+                delegate.onOffsetCommitAck(traceId, authorization);
+            }
+        }
+
```

Comment:
Please review the method ordering of all the inner classes in the factories.

Suggest grouping the `onMqttXxx` and `doMqttXxx` together, also group `onKafkaXxx` and `doKafkaXxx` together, with additional supporting methods at the end of each inner class.